src/main/java/session/FormSession.java

```diff
@@ -237,6 +248,55 @@ private void initialize(boolean newInstance, Map<String, String> sessionData, St
                 CommCareConfigEngine.MINOR_VERSION, storageManager);
         FormplayerSessionWrapper sessionWrapper = new FormplayerSessionWrapper(platform, this.sandbox, sessionData);
         formDef.initialize(newInstance, sessionWrapper.getIIF(), locale, false);
+
+        setVolatilityIndicators();
+    }
+
+    private String getPragma(String key) {
+        String value = formDef.getLocalizer().getText(key);
+        if(value != null) {
+            return formDef.fillTemplateString(
+                    value, TreeReference.rootRef());
+        }
+        return null;
+    }
+
+    private void setVolatilityIndicators()
+    {
+        String volatilityKey = getPragma("Pragma-Volatility-Key");
```

Comment:
Not sure if I understand where these keys are getting set. I have not seen these keys in Form xml, is this a new change on HQ ?

Reply:
Right now, it's manual. I'm adding questions to the form explicitly with these itext keys

Reply:
In the future I think these might get set in the session instead, since otherwise we can't do things like sync **before** loading the form cleanly. Wanted to figure that out before coding in HQ.

Reply:
Just for my understanding this would be like a case ID and `Pragma-Volatility-Entity-Title` would be the case name or something right?

Reply:
Yeah, in the base case. If the form selected more than one thing, those would both need to be more complex to reflect the compound selection, but for most HQ use cases it's just case_id and case_name